central/image/datastore/store/v2/postgres/store_test.go

```diff
@@ -173,68 +157,6 @@ func (s *ImagesStoreSuite) TestNVDCVSS() {
 	protoassert.Equal(s.T(), nvdCvss, imageCve.GetCveBaseInfo().GetCvssMetrics()[0])
 }
 
-func (s *ImagesStoreSuite) TestUpsertLegacyToNew() {
-	image := getTestImage("image1")
-
-	// Upsert image using legacy store. This will insert CVEs and components into the old tables and set created at and
-	// first image occurrence timestamps to current time
-	s.NoError(s.legacyStore.Upsert(s.ctx, image))
-	foundImage, exists, err := s.legacyStore.Get(s.ctx, image.GetId())
-	s.NoError(err)
-	s.True(exists)
-	cloned := image.CloneVT()
-
-	// Reconcile created at and first image occurrence timestamps from the CVEs inserted into the old model
-	for i, comp := range foundImage.GetScan().GetComponents() {
-		for j, vuln := range comp.GetVulns() {
-			cloned.GetScan().GetComponents()[i].GetVulns()[j].FirstSystemOccurrence = vuln.GetFirstSystemOccurrence()
-			cloned.GetScan().GetComponents()[i].GetVulns()[j].FirstImageOccurrence = vuln.GetFirstImageOccurrence()
-		}
-	}
-
-	// Set the created and first image occurrence timestamps in the test image to a future value
-	for _, comp := range image.GetScan().GetComponents() {
-		for _, vuln := range comp.GetVulns() {
-			vuln.FirstSystemOccurrence = protocompat.ConvertTimeToTimestampOrNil(&nextWeek)
-			vuln.FirstImageOccurrence = protocompat.ConvertTimeToTimestampOrNil(&nextWeek)
-		}
-	}
-	// Re-upsert the image into v2 data model store
-	s.NoError(s.store.Upsert(s.ctx, image))
-	foundImage, exists, err = s.legacyStore.Get(s.ctx, image.GetId())
-	s.NoError(err)
-	s.True(exists)
-
-	// Note that we will just compare time fields because the old model can mess up other things like
-	// severity, published time, CVSS, etc. because of over normalization.
-	expectedTimestamps := make(map[string]*timeFields)
-	for _, comp := range cloned.GetScan().GetComponents() {
-		for _, vuln := range comp.GetVulns() {
-			if _, ok := expectedTimestamps[vuln.GetCve()]; !ok {
-				expectedTimestamps[vuln.GetCve()] = &timeFields{
-					createdAt:            vuln.GetFirstSystemOccurrence().AsTime(),
-					firstImageOccurrence: vuln.GetFirstImageOccurrence().AsTime(),
-				}
-			}
-		}
-	}
-
-	actualTimestamps := make(map[string]*timeFields)
-	for _, comp := range foundImage.GetScan().GetComponents() {
-		for _, vuln := range comp.GetVulns() {
-			if _, ok := actualTimestamps[vuln.GetCve()]; !ok {
-				actualTimestamps[vuln.GetCve()] = &timeFields{
-					createdAt:            vuln.GetFirstSystemOccurrence().AsTime(),
-					firstImageOccurrence: vuln.GetFirstImageOccurrence().AsTime(),
```

Comment:
Would ROX-29911 also add `FirstImageOccurrence` time? Wouldn't that make the lookup table pretty big?